File: src/test/java/kitchenpos/eatinorders/domain/FakeOrderTableRepository.java
package kitchenpos.eatinorders.domain;

import kitchenpos.eatinorders.tobe.domain.entity.OrderTable;
import kitchenpos.eatinorders.tobe.domain.repository.OrderTableRepository;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

public class FakeOrderTableRepository implements OrderTableRepository {
    private final Map<UUID, OrderTable> repository = new HashMap<>();

    @Override
    public OrderTable save(OrderTable orderTable) {
        repository.put(orderTable.getId(), orderTable);
        return repository.get(orderTable.getId());
    }

    @Override
    public Optional<OrderTable> findBy(UUID id) {
        return Optional.ofNullable(repository.get(id));
    }
}

File: src/test/kotlin/kitchenpos/application/fake/FakeOrderTableRepository.kt
package kitchenpos.application.fake

import kitchenpos.order.eatin.domain.OrderTable
import kitchenpos.order.eatin.domain.OrderTableRepository
import java.util.*

class FakeOrderTableRepository : OrderTableRepository {
    private val orderTables = mutableMapOf<UUID, OrderTable>()

    override fun save(entity: OrderTable?): OrderTable {
        orderTables[entity!!.id] = entity
        return entity
    }

    override fun findById(id: UUID?): Optional<OrderTable> {
        return Optional.ofNullable(orderTables[id])
    }

    override fun findAll(): MutableList<OrderTable> {
        return orderTables.values.toMutableList()
    }
}
